mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -548,6 +615,20 @@ def _compare_metrics(metric_a, metric_b):
                 metric_b.value,
             )
 
+        logged_metric = None
+
+        if logged_metrics is not None:
+            for m in logged_metrics:
+                if logged_metric is None or _compare_metrics(m, logged_metric):
+                    logged_metric = m
+
+        if logged_metric is None:
+            raise MlflowException(
+                "Invalid parameters for _update_latest_metric_if_necessary. "
+                "Please set logged_metrics",
+                INVALID_PARAMETER_VALUE,
```

Comment:
Actually, the error will never arise due to previous checks in log_metric(s). `_update_latest_metric_if_necessary` is only called when it has at least one metric so that this case never occurs. Should when then switch back to just returning? Or do you have another recommendation?